pkg/proxy/proxier.go

```diff
@@ -404,6 +466,14 @@ func iptablesInit(ipt iptables.Interface) error {
 	if _, err := ipt.EnsureChain(iptables.TableNAT, iptablesContainerPortalChain); err != nil {
 		return err
 	}
+
+	// TODO (comment from thockin): This is a little scary - if somehow these rules get out of order, this will consume
```

Comment:
I stand by this comment - I want to play a bit more to come up with a suggested implementation.  Will have time later tonight.